bleak/backends/winrt/client.py

```diff
@@ -0,0 +1,824 @@
+            )
+        )
+
+        # Start a GATT Session to connect
+        event = asyncio.Event()
+        self._connect_events.append(event)
+        try:
+            self._session = await GattSession.from_device_id_async(
+                self._requester.bluetooth_device_id
+            )
+            # This keeps the device connected until we dispose the session or
+            # until we set MaintainConnection = False.
+            self._session.maintain_connection = True
+            await asyncio.wait_for(event.wait(), timeout=timeout)
+        except BaseException:
+            handle_disconnect()
+            raise
+        finally:
+            self._connect_events.remove(event)
+
+        # Obtain services, which also leads to connection being established.
+        await self.get_services(use_cached=use_cached)
+
+        return True
+
+    async def disconnect(self) -> bool:
+        """Disconnect from the specified GATT server.
+
+        Returns:
+            Boolean representing if device is disconnected.
+
+        """
+        logger.debug("Disconnecting from BLE device...")
+        # Remove notifications. Remove them first in the BleakBridge and then clear
+        # remaining notifications in Python as well.
+        for handle, (fcn, fcn_token) in list(self._notification_callbacks.items()):
+            char = self.services.get_characteristic(handle)
+            char.obj.remove_value_changed(fcn_token)
+        self._notification_callbacks.clear()
+
+        # Dispose all service components that we have requested and created.
+        for service in self.services:
+            service.obj.close()
+        self.services = BleakGATTServiceCollection()
+        self._services_resolved = False
+
+        # Without this, disposing the BluetoothLEDevice won't disconnect it
+        if self._session:
+            self._session.close()
+
+        # Dispose of the BluetoothLEDevice and see that the connection
+        # status is now Disconnected.
+        if self._requester:
+            event = asyncio.Event()
+            self._disconnect_events.append(event)
+            try:
+                self._requester.close()
+                await asyncio.wait_for(event.wait(), timeout=10)
+            finally:
+                self._disconnect_events.remove(event)
+
+        return True
+
+    async def is_connected(self) -> bool:
+        """Check connection status between this client and the server.
+
+        Returns:
+            Boolean representing connection status.
+
+        """
+        if self._requester:
+            return (
+                self._requester.connection_status == BluetoothConnectionStatus.CONNECTED
+            )
+        else:
+            return False
+
+    async def pair(self, protection_level=None, **kwargs) -> bool:
+        """Attempts to pair with the device.
+
+        Keyword Args:
+            protection_level:
+                    DevicePairingProtectionLevel
```

Comment:
Is this a `winrt` type? Maybe use the full namespace here to make it clear. Or use strings instead so that the code can work cross-platform.

Reply:
Yes. Added complete namespace both here and in `dotnet` backend.